### ISSUE TITLE
Error when registering

### ISSUE THREAD

**Issue by Tostino** [PrefectHQ/prefect-sqlalchemy#28]
I am getting an error when attempting to register this modules credentials as a Block in my Orion server.

Prefect 2.2.0 
Prefect-sqlalchemy 0.1.2

See logs:
```
root@workflow:/opt/workflow# prefect block register --module prefect_sqlalchemy.credentials
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/prefect/cli/_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/asyncutils.py", line 193, in wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/asyncutils.py", line 140, in run_async_in_new_loop
    return anyio.run(partial(__fn, *args, **kwargs))
  File "/usr/local/lib/python3.9/dist-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/usr/local/lib/python3.9/dist-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.9/dist-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/usr/local/lib/python3.9/dist-packages/prefect/cli/block.py", line 118, in register
    imported_module = import_module(name=module_name)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/usr/local/lib/python3.9/dist-packages/prefect_sqlalchemy/__init__.py", line 2, in <module>
    from .credentials import DatabaseCredentials, AsyncDriver, SyncDriver  # noqa
  File "/usr/local/lib/python3.9/dist-packages/prefect_sqlalchemy/credentials.py", line 91, in <module>
    class DatabaseCredentials(Block):
  File "pydantic/main.py", line 283, in pydantic.main.ModelMetaclass.__new__
  File "/usr/lib/python3.9/abc.py", line 85, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/dispatch.py", line 99, in _register_subclass_of_base_type
    register_type(cls)
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/dispatch.py", line 154, in register_type
    key = get_dispatch_key(cls)
  File "/usr/local/lib/python3.9/dist-packages/prefect/utilities/dispatch.py", line 76, in get_dispatch_key
    dispatch_key = dispatch_key()
  File "/usr/local/lib/python3.9/dist-packages/prefect/blocks/core.py", line 199, in __dispatch_key__
    return block_schema_to_key(cls._to_block_schema())
  File "/usr/local/lib/python3.9/dist-packages/prefect/blocks/core.py", line 346, in _to_block_schema
    fields = cls.schema()
  File "pydantic/main.py", line 664, in pydantic.main.BaseModel.schema
  File "pydantic/schema.py", line 186, in pydantic.schema.model_schema
  File "pydantic/schema.py", line 580, in pydantic.schema.model_process_schema
  File "pydantic/schema.py", line 621, in pydantic.schema.model_type_schema
  File "pydantic/schema.py", line 254, in pydantic.schema.field_schema
  File "pydantic/schema.py", line 526, in pydantic.schema.field_type_schema
  File "pydantic/schema.py", line 847, in pydantic.schema.field_singleton_schema
  File "pydantic/schema.py", line 742, in pydantic.schema.field_singleton_sub_fields_schema
  File "pydantic/schema.py", line 526, in pydantic.schema.field_type_schema
  File "pydantic/schema.py", line 889, in pydantic.schema.field_singleton_schema
  File "pydantic/schema.py", line 594, in pydantic.schema.model_process_schema
  File "/usr/local/lib/python3.9/dist-packages/prefect/blocks/core.py", line 118, in schema_extra
    schema["block_type_slug"] = model.get_block_type_slug()
AttributeError: type object 'URL' has no attribute 'get_block_type_slug'
An exception occurred.
```

**Comment by Tostino** [comment]
Well, same error seems to be happening when trying to generate a deployment build script as well.

**Comment by Tostino** [comment]
Just FYI, I tried allowing arbitrary types through Pydantic by changing the configuration.py like so: https://github.com/Tostino/prefect-sqlalchemy/commit/3413f15168c638e42b7cf68278387a2b99de9951

Did not work.

**Comment by desertaxle** [comment]
Thanks for the report @Tostino! It looks like we had an incompatible type for the `url` attribute. This will be fixed once #29 is merged and released.
